csrc/fusion_segmenter.cpp

```diff
@@ -1489,6 +1489,32 @@ void convertInputRfactorsToRoots(Fusion* fusion) {
     NVF_ERROR(new_root_domain.size() == tv->domain()->contiguity().size());
     auto new_td = IrBuilder::create<TensorDomain>(
         new_root_domain, tv->domain()->contiguity());
+
```

Comment:
I wonder if we really need this function at all. Some code may be naively assuming fusion inputs do not have rfactor domains, but I don't think that's necessary. Maybe just try disabling this function entirely?

Reply:
I guess I can try removing this one and see if the world would just crash and burn...

Reply:
Thanks for the cleanup! 😉